src/github.com/getlantern/pac-cmd/linux.c

```diff
@@ -26,17 +27,26 @@ int togglePac(bool turnOn, const char* pacUrl)
     }
   }
   else {
+    if (strlen(pacUrl) != 0) {
+      // clear pac setting only if it's equal to pacUrl
+      char* old_mode = g_settings_get_string(setting, "mode");
+      char* old_pac_url = g_settings_get_string(setting, "autoconfig-url");
+      if (strcmp(old_mode, "auto") != 0 || old_pac_url == NULL || strcmp(old_pac_url, pacUrl) != 0 ) {
```

Comment:
Good catch @atavism . No official document found but from [this thread](https://mail.gnome.org/archives/commits-list/2010-September/msg01431.html) `g_settings_get_string` will never return NULL, also confirmed from [source code](https://github.com/GNOME/glib/blob/6cd1f8b40ff66509eedfce9c3c46cbef799a7ffe/gio/gsettings.c#L1236). I'll remove following checking against NULL.